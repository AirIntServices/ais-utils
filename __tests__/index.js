const { groupBy, oldGroupBy } = require('../index');

describe('Testing idempotence of groupBy and fastGroupBy', () => {
  test('behaves the same when passing empty array', () => {
    const arr = [];
    expect(groupBy(arr)).toEqual(oldGroupBy(arr));
  });

  test('behaves the same for grouping simple array', () => {
    const arr = [
      { foo: 1, bar: 'A' },
      { foo: 1, bar: 'B' },
      { foo: 2, bar: 'C' },
    ];
    const fieldName = 'foo';
    expect(groupBy(arr, fieldName)).toEqual(oldGroupBy(arr, fieldName));
  });

  test('behaves the same when grouping by non existing field', () => {
    const arr = [
      { foo: 1, bar: 'A' },
      { foo: 1, bar: 'B' },
      { foo: 2, bar: 'C' },
    ];
    const fieldName = 'baz';
    expect(groupBy(arr, fieldName)).toEqual(oldGroupBy(arr, fieldName));
  });
});
