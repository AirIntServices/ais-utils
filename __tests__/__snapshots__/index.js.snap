// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sortByFields should ignore nonexisting parameters 1`] = `
Array [
  Object {
    "line": 3,
    "page": 2,
    "test": "B",
  },
  Object {
    "line": 1,
    "page": 1,
    "test": "a",
  },
  Object {
    "line": 2,
    "page": 1,
    "test": "A",
  },
  Object {
    "line": 4,
    "page": 1,
    "test": 2,
  },
  Object {
    "line": 4,
    "page": 1,
    "test": 1,
  },
]
`;

exports[`sortByFields should ignore null direction 1`] = `
Array [
  Object {
    "line": 1,
    "page": 1,
    "test": "a",
  },
  Object {
    "line": 2,
    "page": 1,
    "test": "A",
  },
  Object {
    "line": 3,
    "page": 2,
    "test": "B",
  },
  Object {
    "line": 4,
    "page": 1,
    "test": 2,
  },
  Object {
    "line": 4,
    "page": 1,
    "test": 1,
  },
]
`;

exports[`sortByFields should return sort by page ASC 1`] = `
Array [
  Object {
    "line": 1,
    "page": 1,
    "test": "a",
  },
  Object {
    "line": 2,
    "page": 1,
    "test": "A",
  },
  Object {
    "line": 4,
    "page": 1,
    "test": 2,
  },
  Object {
    "line": 4,
    "page": 1,
    "test": 1,
  },
  Object {
    "line": 3,
    "page": 2,
    "test": "B",
  },
]
`;

exports[`sortByFields should return sort by page ASC 2`] = `
Array [
  Object {
    "line": 1,
    "page": 1,
    "test": "a",
  },
  Object {
    "line": 2,
    "page": 1,
    "test": "A",
  },
  Object {
    "line": 4,
    "page": 1,
    "test": 2,
  },
  Object {
    "line": 4,
    "page": 1,
    "test": 1,
  },
  Object {
    "line": 3,
    "page": 2,
    "test": "B",
  },
]
`;

exports[`sortByFields should return sort by page ASC AND line ASC 1`] = `
Array [
  Object {
    "line": 1,
    "page": 1,
    "test": "a",
  },
  Object {
    "line": 2,
    "page": 1,
    "test": "A",
  },
  Object {
    "line": 4,
    "page": 1,
    "test": 2,
  },
  Object {
    "line": 4,
    "page": 1,
    "test": 1,
  },
  Object {
    "line": 3,
    "page": 2,
    "test": "B",
  },
]
`;

exports[`sortByFields should return sort by page ASC AND line ASC 2`] = `
Array [
  Object {
    "line": 1,
    "page": 1,
    "test": "a",
  },
  Object {
    "line": 2,
    "page": 1,
    "test": "A",
  },
  Object {
    "line": 4,
    "page": 1,
    "test": 2,
  },
  Object {
    "line": 4,
    "page": 1,
    "test": 1,
  },
  Object {
    "line": 3,
    "page": 2,
    "test": "B",
  },
]
`;

exports[`sortByFields should return sort by page ASC AND line DESC 1`] = `
Array [
  Object {
    "line": 4,
    "page": 1,
    "test": 2,
  },
  Object {
    "line": 4,
    "page": 1,
    "test": 1,
  },
  Object {
    "line": 2,
    "page": 1,
    "test": "A",
  },
  Object {
    "line": 1,
    "page": 1,
    "test": "a",
  },
  Object {
    "line": 3,
    "page": 2,
    "test": "B",
  },
]
`;

exports[`sortByFields should return sort by page DESC 1`] = `
Array [
  Object {
    "line": 3,
    "page": 2,
    "test": "B",
  },
  Object {
    "line": 1,
    "page": 1,
    "test": "a",
  },
  Object {
    "line": 2,
    "page": 1,
    "test": "A",
  },
  Object {
    "line": 4,
    "page": 1,
    "test": 2,
  },
  Object {
    "line": 4,
    "page": 1,
    "test": 1,
  },
]
`;

exports[`sortByFields should return sort by page DESC AND line ASC 1`] = `
Array [
  Object {
    "line": 3,
    "page": 2,
    "test": "B",
  },
  Object {
    "line": 1,
    "page": 1,
    "test": "a",
  },
  Object {
    "line": 2,
    "page": 1,
    "test": "A",
  },
  Object {
    "line": 4,
    "page": 1,
    "test": 2,
  },
  Object {
    "line": 4,
    "page": 1,
    "test": 1,
  },
]
`;

exports[`sortByFields should return sort by page DESC AND line DESC 1`] = `
Array [
  Object {
    "line": 3,
    "page": 2,
    "test": "B",
  },
  Object {
    "line": 4,
    "page": 1,
    "test": 2,
  },
  Object {
    "line": 4,
    "page": 1,
    "test": 1,
  },
  Object {
    "line": 2,
    "page": 1,
    "test": "A",
  },
  Object {
    "line": 1,
    "page": 1,
    "test": "a",
  },
]
`;

exports[`sortByFields should return sort by test DESC AND case sensitive 1`] = `
Array [
  Object {
    "line": 1,
    "page": 1,
    "test": "a",
  },
  Object {
    "line": 3,
    "page": 2,
    "test": "B",
  },
  Object {
    "line": 2,
    "page": 1,
    "test": "A",
  },
  Object {
    "line": 4,
    "page": 1,
    "test": 2,
  },
  Object {
    "line": 4,
    "page": 1,
    "test": 1,
  },
]
`;

exports[`sortByFields should return sort by test DESC AND not case sensitive 1`] = `
Array [
  Object {
    "line": 3,
    "page": 2,
    "test": "B",
  },
  Object {
    "line": 1,
    "page": 1,
    "test": "a",
  },
  Object {
    "line": 2,
    "page": 1,
    "test": "A",
  },
  Object {
    "line": 4,
    "page": 1,
    "test": 2,
  },
  Object {
    "line": 4,
    "page": 1,
    "test": 1,
  },
]
`;
